include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_scan.h

```diff
@@ -219,40 +182,23 @@ single_pass_scan_impl(sycl::queue __queue, _InRange&& __in_rng, _OutRange&& __ou
             auto in_end = __in_rng.begin() + next_offset;
             auto out_begin = __out_rng.begin() + current_offset;
 
-
-#if SCAN_KT_DEBUG
-            debug3[tile_id] = current_offset;
-            debug4[tile_id] = next_offset;
-#endif
-
             if (current_offset >= n)
                 return;
 
             auto local_sum = sycl::joint_reduce(group, in_begin, in_end, __binary_op);
-#if SCAN_KT_DEBUG
-            debug1[tile_id] = local_sum;
-#endif
-
             _Type prev_sum = 0;
 
             // The first sub-group will query the previous tiles to find a prefix
             if (subgroup.get_group_id() == 0)
             {
-                __scan_status_flag<_Type> flag(status_flags, tile_id);
+                __scan_status_flag<_Type> flag(status_flags, tile_id, sums, status_flags_size);
```

Comment:
Maybe its a nit but the order of parameters to this function doesn't seem the most intuitive. Maybe `(tile_id, status_flags, tile_sums, status_flags_size)`?

Reply:
I've reordered as you suggested.